src/codec_aom.c

```diff
@@ -972,6 +972,7 @@ static avifResult aomCodecEncodeImage(avifCodec * codec,
             bps = 16;
         }
         aomImage.bps = bps;
+        // See avifImageCalcAOMFmt(). libaom doesn't have AOM_IMG_FMT_I400, so we use AOM_IMG_FMT_I420 as a substitute for monochrome.
```

Comment:
I am sorry my previous comment about the code here wasn't clear. Please change the next two lines to what you had before:

```
        aomImage.x_chroma_shift = (alpha || codec->internal->formatInfo.monochrome) ? 1 : codec->internal->formatInfo.chromaShiftX;
        aomImage.y_chroma_shift = (alpha || codec->internal->formatInfo.monochrome) ? 1 : codec->internal->formatInfo.chromaShiftY;
```

The reason is that what you had before matches the code for alpha and AVIF_PIXEL_FORMAT_YUV400 in `avifImageCalcAOMFmt()`, so the new comment we added explains that's why we set both `x_chroma_shift` and `y_chroma_shift` to 1 if `(alpha || codec->internal->formatInfo.monochrome)` is true.